src/parse.cpp

```diff
@@ -0,0 +1,76 @@
+#include "lib/parse.h"
+#include "lib/lex.h"
+
+TreeLeaf::TreeLeaf (double val) {
+    value = val; // initializing one numerical
+};
```

Comment:
Is it possible to stick to a format of 1 newline between functions under the same namespace, and 3 newlines between functions of different namespace (the numbers can be different I would just prefer them to be further spaced)